libs/@local/repo-chores/scripts/prepublish.ts

```diff
@@ -0,0 +1,96 @@
+import path from "node:path";
+
+import chalk from "chalk";
+import execa from "execa";
+import fs from "fs-extra";
+
+import { UserFriendlyError } from "./shared/errors";
+import { checkIfDirHasUncommittedChanges } from "./shared/git";
+import {
+  derivePackageInfoFromEnv,
+  outputPackageInfo,
+} from "./shared/package-infos";
+import { updateJson } from "./shared/update-json";
+
+const script = async () => {
+  console.log(chalk.bold("Cleaning up before publishing..."));
+
+  const packageInfo = await derivePackageInfoFromEnv();
+  outputPackageInfo(packageInfo);
+
+  if (await checkIfDirHasUncommittedChanges(packageInfo.path)) {
+    throw new UserFriendlyError(
+      `Please commit or revert changes in ${packageInfo.path} before running this script`,
+    );
+  }
+
+  if (!(await fs.pathExists(path.resolve(packageInfo.path, ".npmignore")))) {
+    throw new UserFriendlyError(
+      `Please create .npmignore in ${packageInfo.path} before running this script`,
+    );
+  }
+
+  process.stdout.write(`Removing dist...`);
+
+  await fs.remove(path.resolve(packageInfo.path, "dist"));
+
+  process.stdout.write(" Done\n");
+  process.stdout.write(`Building...`);
+
+  // tsconfig.json is supposed to configured for local development and linting.
+  // We need to override some options to generate a build that is ready for publishing.
+  await execa(
+    "tsc",
+    // prettier-ignore
+    [
+      "--project", "tsconfig.json",
+      
+      "--declaration", "true",
+      "--jsx", "react-jsx",
+      "--noEmit", "false",
+      "--outDir", "dist",
+      "--target", "es2020",
+    ],
+    {
+      cwd: packageInfo.path,
+      stdout: "inherit",
+    },
+  );
+
+  process.stdout.write(" Done\n");
+
+  process.stdout.write(`Updating package.json...`);
+
+  await updateJson(
+    path.resolve(packageInfo.path, "package.json"),
+    (packageJson) => {
+      /* eslint-disable @typescript-eslint/no-unsafe-member-access,no-param-reassign -- see comment on updateJson() for potential improvement */
+      if (packageJson.main !== "src/main.ts") {
+        throw new UserFriendlyError(
+          "Unexpected value for field `main` in `package.json`. Please align this package with other publishable packages for consistency",
+        );
+      }
```

Comment:
I would add a bit more detail to make this easier to resolve if anyone ever hits it (same applies below)
```suggestion
      const expectedMainName = "src/main.ts";
      if (packageJson.main !== expectedMainName) {
        throw new UserFriendlyError(
          `Unexpected value for field 'main' in 'package.json'. Please align this package with other publishable packages for consistency. Expected: '${expectedMainName}'. Got: '${packageJson.main}'`,
        );
      }
```

Reply:
https://github.com/hashintel/hash/pull/2005/commits/31d29050c9b52eafa9a4288a7c12d3596cc805e5